src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -634,6 +656,12 @@ public override BoundNode VisitUnaryOperator(BoundUnaryOperator node)
             CheckUnsafeType(node);
             CheckLiftedUnaryOp(node);
             CheckDynamic(node);
+
+            if (_inExpressionLambda && node.MethodOpt is MethodSymbol method && method.IsAbstract && method.IsStatic)
```

Comment:
>method.IsAbstract && method.IsStatic [](start = 80, length = 36)

Consider extracting a helper `IsValidCallInExpressionTree` or something like that and using it in all these locations.